pandas/core/array_algos/take.py

```diff
@@ -360,7 +360,14 @@ def wrapper(
         if out_dtype is not None:
             out = out.view(out_dtype)
         if fill_wrap is not None:
+            # FIXME: if we get here with dt64/td64 we need to be sure we have
+            #  matching resos
+            if fill_value.dtype.kind == "m":
```

Comment:
Might be worth making a re-usable function for this? I could see this useful in other areas. Something like `reso_for_type`